flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1352,52 +1359,6 @@ public void run() {
 		}
 	}
 
-	/**
-	 * Discards the given pending checkpoint because of the given cause.
-	 *
-	 * @param pendingCheckpoint to discard
-	 * @param cause for discarding the checkpoint
-	 * @param executionAttemptID the execution attempt id of the failing task.
-	 */
-	private void discardCheckpoint(
-		PendingCheckpoint pendingCheckpoint,
-		@Nullable Throwable cause,
-		ExecutionAttemptID executionAttemptID) {
-		assert(Thread.holdsLock(lock));
-		Preconditions.checkNotNull(pendingCheckpoint);
-
-		final long checkpointId = pendingCheckpoint.getCheckpointId();
-
-		LOG.info("Discarding checkpoint {} of job {}.", checkpointId, job, cause);
-
-		if (cause == null) {
-			failPendingCheckpointDueToTaskFailure(pendingCheckpoint, CheckpointFailureReason.CHECKPOINT_DECLINED, executionAttemptID);
-		} else if (cause instanceof CheckpointException) {
-			CheckpointException exception = (CheckpointException) cause;
-			failPendingCheckpointDueToTaskFailure(pendingCheckpoint, exception.getCheckpointFailureReason(), cause, executionAttemptID);
-		} else {
-			failPendingCheckpointDueToTaskFailure(pendingCheckpoint, CheckpointFailureReason.JOB_FAILURE, cause, executionAttemptID);
-		}
-
-		rememberRecentCheckpointId(checkpointId);
-
-		// we don't have to schedule another "dissolving" checkpoint any more because the
-		// cancellation barriers take care of breaking downstream alignments
-		// we only need to make sure that suspended queued requests are resumed
-
-		boolean haveMoreRecentPending = false;
-		for (PendingCheckpoint p : pendingCheckpoints.values()) {
-			if (!p.isDiscarded() && p.getCheckpointId() >= pendingCheckpoint.getCheckpointId()) {
-				haveMoreRecentPending = true;
-				break;
-			}
-		}
-
-		if (!haveMoreRecentPending) {
-			triggerQueuedRequests();
-		}
```

Comment:
This if check is gone now? Now this is called always?

Reply:
Please see the response below. I changed the behavior a bit of resuming periodic triggering.